test/util/TestFileUtils.cpp

```diff
@@ -68,6 +68,19 @@ TEST(FileUtils, handlesEmptyPathToExtractExpandedPath)
     ASSERT_STREQ("", extendedPath.c_str());
 }
 
+TEST(FileUtils, handlesIllegalCharacterInExtractExpandedPath)
+{
+    ASSERT_THROW(FileUtils::ExtractExpandedPath("|"), wordexp_fail_error);
```

Comment:
Google test provides support for Value-Parameterized tests which are a better match for this type of test. https://github.com/google/googletest/blob/main/docs/advanced.md#value-parameterized-tests